The array used is 1 to small, so increase the size to avoid a buffer overflow.
See https://github.com/open-mpi/ompi/issues/9018

Author: Alexander Grund (TU Dresden)
diff --git a/opal/mca/common/ofi/common_ofi.c b/opal/mca/common/ofi/common_ofi.c
index ff73e6f069..a5a200d205 100644
--- a/opal/mca/common/ofi/common_ofi.c
+++ b/opal/mca/common/ofi/common_ofi.c
@@ -304,7 +304,7 @@ static uint32_t get_package_rank(int32_t num_local_peers, uint16_t my_local_rank
     int i;
     uint16_t relative_locality;
     uint16_t current_package_rank = 0;
-    uint16_t package_ranks[num_local_peers];
+    uint16_t package_ranks[num_local_peers + 1];
     opal_process_name_t pname;
     opal_status_t rc;
     char **peers = NULL;
